apps/ejabberd/src/ejabberd_auth_internal.erl

```diff
@@ -56,11 +56,11 @@
 
 -include("ejabberd.hrl").
 
--record(passwd, {us :: ejabberd:simple_bare_jid(),
-                 password :: binary()
+-record(passwd, {us :: ejabberd:simple_bare_jid() | atom(), % atom() for mnesia:dirty_select
```

Comment:
Great! A short comment in the code explaining why we do this will be helpful in future :)